### PR TITLE
feat(agents-api,migrations): Add trigram search to FTS

### PR DESCRIPTION
**Pull request by creatorrr**
### **User description**
- **refactor(agents-api): Refactor text_to_tsquery -> text_to_keywords**
- **feat(agents-api,migrations): Add trigram search to FTS**

---
# EntelligenceAI PR Summary 
 **Purpose**:
- Enhance keyword extraction and document search functionalities for improved accuracy and flexibility.

**Changes**:
- **Enhancement**: Refined `extract_keywords` for better accuracy by adjusting excluded labels and filtering single-character keywords.
- **New Feature**: Introduced `text_to_keywords` for enhanced search capabilities and implemented trigram similarity with `english_unaccent` configuration for improved search functionality.
- **Test**: Updated tests to ensure alignment and effectiveness of new features.

**Impact**:
- Users will experience more precise keyword extraction and flexible, accurate text searches, enhancing overall user experience.

___

### **PR Type**
Enhancement, Tests, Bug fix

___

### **Description**
- Introduced trigram-based similarity for improved text search.

- Enhanced keyword extraction with `text_to_keywords` function.

- Updated database migrations for custom text search configurations.

- Refactored and improved error handling in language validation.

- Added and updated tests for new functionalities and edge cases.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>9 files</summary><table>
<tr>
  <td><strong>nlp.py</strong><dd><code>Refactored keyword extraction and added `text_to_keywords`.</code></dd></td>
  <td><a href="https://github.com/julep-ai/julep/pull/1210/files#diff-66deb57e966d8a3ac8b61fe3240fff24ef1d0aa2f31e9303c88df372245c3be0">+14/-12</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>get_doc_search.py</strong><dd><code>Improved language validation and search parameter handling.</code></dd></td>
  <td><a href="https://github.com/julep-ai/julep/pull/1210/files#diff-6cd50924db8f0ca2e89a951db3be7cf2cae078d5c6d7d100d3abea2f0cdb79d3">+21/-18</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>gather_messages.py</strong><dd><code>Integrated keyword extraction for conversation snippet queries.</code></dd></td>
  <td><a href="https://github.com/julep-ai/julep/pull/1210/files#diff-1b7f13a02e66bbfa4c5963a5d6339d096f0cfcf5cdaeb414fac26c1e6189f4a0">+6/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>search_docs_by_text.py</strong><dd><code>Enhanced text search with trigram similarity and keyword extraction.</code></dd></td>
  <td><a href="https://github.com/julep-ai/julep/pull/1210/files#diff-d8e9417b5496c44804e81c8ed8f8dd289ad387e5a7595d786987b2c97adb050c">+22/-14</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>search_docs_hybrid.py</strong><dd><code>Updated hybrid search to include trigram similarity.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/julep-ai/julep/pull/1210/files#diff-6295463a14ae8b5e675326a1737390965f4dbd1be0614371510d36904a514f78">+15/-6</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>000030_add_unaccent_search_config.down.sql</strong><dd><code>Reverted custom text search configuration changes.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/julep-ai/julep/pull/1210/files#diff-3bea8349c24f140cff01cf1a16f638a67ab24bfbec881b412926862d4624e3bb">+84/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>000030_add_unaccent_search_config.up.sql</strong><dd><code>Added custom text search configuration with unaccent support.</code></dd></td>
  <td><a href="https://github.com/julep-ai/julep/pull/1210/files#diff-6060acaa44c4fc69501102cb10ae279410f138e9471a22ca64cb43adf972e005">+92/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>000031_add_trigram_search.down.sql</strong><dd><code>Reverted trigram-based search enhancements.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/julep-ai/julep/pull/1210/files#diff-fb48f4e5225a659fa1be175483dfc9494915143858b8f1de1b298a8fa5fa3440">+194/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>000031_add_trigram_search.up.sql</strong><dd><code>Introduced trigram-based search and hybrid search enhancements.</code></dd></td>
  <td><a href="https://github.com/julep-ai/julep/pull/1210/files#diff-59dd3bf97cdf26d50553b8d03e1593909d7278d8ff9341ebdd6854c5b4d15882">+319/-0</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Bug fix</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>search_docs.py</strong><dd><code>Adjusted response time calculation in document search.</code>&nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/julep-ai/julep/pull/1210/files#diff-7c78741623ef43e8990dbf7bc74bf97e64950e60d60eae1249afb33eb38083af">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>test_get_doc_search.py</strong><dd><code>Added tests for language validation and search parameter handling.</code></dd></td>
  <td><a href="https://github.com/julep-ai/julep/pull/1210/files#diff-8d711571115dd4b7f5da1b5a7f2ca7cbc39ed31c220a57465c88dc6b2ca3b151">+23/-15</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>test_nlp_utilities.py</strong><dd><code>Updated tests for `text_to_keywords` and keyword extraction.</code></dd></td>
  <td><a href="https://github.com/julep-ai/julep/pull/1210/files#diff-c5d1cfb3d8c78c573a6228a7773e901550d5ef68216991a66d0bdc9c261f8952">+42/-46</a>&nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

----

> [!IMPORTANT]
> Enhances FTS with trigram search, refines keyword extraction, and updates configurations and tests across `agents-api` and `memory-store`.
> 
>   - **Behavior**:
>     - Adds trigram search to FTS in `search_docs_by_text.py` and `search_docs_hybrid.py`.
>     - Introduces `text_to_keywords` in `nlp.py` for keyword extraction.
>     - Updates `get_language()` in `get_doc_search.py` to return `english_unaccent` for English.
>   - **Migrations**:
>     - Adds `000030_add_unaccent_search_config.up.sql` and `000031_add_trigram_search.up.sql` for new search configurations.
>     - Provides down migrations to revert changes in `000030_add_unaccent_search_config.down.sql` and `000031_add_trigram_search.down.sql`.
>   - **Tests**:
>     - Updates tests in `test_get_doc_search.py` and `test_nlp_utilities.py` to cover new functionalities and configurations.
>   - **Misc**:
>     - Refactors `text_to_tsvector_query` to `text_to_keywords` in `nlp.py`.
>     - Adjusts `get_search_fn_and_params()` in `get_doc_search.py` to handle new search parameters.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=julep-ai%2Fjulep&utm_source=github&utm_medium=referral)<sup> for b30c659c63b51c300345d81e68b903f06fa4d5df. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: agents-api/agents_api/queries/docs/search_docs_by_text.py [hunk 1]
```diff
@@ -61,8 +64,11 @@ async def search_docs_by_text(
     # Extract owner types and IDs
     owner_types: list[str] = [owner[0] for owner in owners]
     owner_ids: list[str] = [str(owner[1]) for owner in owners]
-    #  Pre-process rawtext query
-    query = text_to_tsvector_query(query, split_chunks=True)
+
+    # Pre-process rawtext query if too long or is_conversation_snippet is True
+    if len(query) > 300 or is_conversation_snippet:
```
Comment (whiterabbit1983) [inline agents-api/agents_api/queries/docs/search_docs_by_text.py:69]: Magic number
